### PR TITLE
add release workflow for dataframe api

### PR DESCRIPTION
**Pull request by MarcoGorelli**
As discussed in the last call, let's start publishing the dataframe-api package to PyPI

This means consumers will be able to type-check their usage of the dataframe api, and this will tell them when they're staying within the standard and when they're not

### CODE REVIEW

File: .github/workflows/publish-to-pypi.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        build
+        --user
+    - name: Build a binary wheel and a source tarball
+      run: >-
+        python3 -m
+        build
+        --sdist
+        --wheel
+        --outdir dist/
+        spec/API_specification
```
Comment (rgommers) [inline .github/workflows/publish-to-pypi.yml:30]: minor: these would be much easier to read as one-liners. E.g.:
```
run: |
  python3 -m pip install build
```

File: .github/workflows/publish-to-pypi.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      uses: actions/setup-python@v4
+      with:
+        python-version: "3.x"
+
+    - name: Install pypa/build
+      run: >-
+        python3 -m
+        pip install
+        build
+        --user
```
Comment (rgommers) [inline .github/workflows/publish-to-pypi.yml:22]: minor: `--user` is basically obsolete, and should not be needed in CI; you can remove it here I'd say.

File: .github/workflows/publish-to-pypi.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
+name: Publish Python 🐍 distributions 📦 to PyPI and TestPyPI
+
+on: push
```
Comment (rgommers) [inline .github/workflows/publish-to-pypi.yml:3]: The conditions don't seem quite right here. It only runs on `push`, not on `pull`. And then the publish stage only publishes on any tags. Should this be something like:
```yaml
on:
  pull_request:
    branches:
      - main
  push:
    tags:
      - 'v*'
``` 
?

**Review by rgommers — COMMENTED** [review]
Thanks Marco. This looks pretty good to me - a few comments. The deploy token exists I see, so this looks about good to go.

**Comment by MarcoGorelli** [comment]
closing just to keep my open PRs down, but others are welcome to take this forward

**Pull request closed without merging**